In [1]:
import numpy as np
import pandas as pd

data = pd.read_json('data.json')
# df = pd.DataFrame(data)

# data

In [2]:
# We'll access restaurants and do a function on each object and replace it
df = pd.DataFrame(data)
ratings_array = []
def expenses(val):
    print(val)
    obj = val
    # obj = obj['restaurants']
    cost = 0
    cnt = 0
    if(type(obj) == float):
        # NaN value preprocessed
        return 0
    
    if('rating' in obj):
        lc_obj = {"name" : obj['name'], "rating": 0 , "address" : "Not in data"}
        try:
            lc_obj['rating'] = float(obj['rating'])
            cnt = cnt+1
            ratings_array.append(lc_obj)
            
        except:
            lc_obj = lc_obj
        
        try:
            lc_obj['address'] = obj['address']
        except:
            lc_obj = lc_obj
        
        return lc_obj
    else:
       return np.nan
        
    
# 12 problem 
df2 = df
df2 = df2['Bangalore']
df2 = df2.iloc[0]
df2 = pd.DataFrame(df2)
# print(df2)
df2 = df2['restaurants'].apply(expenses)

{'name': 'Go italia', 'type': 'Pizzas, Pastas, Continental, Italian, Salads, Juices, Desserts', 'rating': '4.1', 'price': '₹500 FOR TWO', 'link': 'go-italia-hsr-layout-bangalore-101748'}
{'name': 'Fiero! Pizzeria', 'type': 'Pizzas, Italian, Pastas, Desserts, Beverages', 'rating': '4.1', 'price': '₹400 FOR TWO', 'link': 'fiero-pizzeria-koramangala-bangalore-101751'}
{'name': 'The Bier Library', 'type': 'Continental, North Indian, Pizzas, American, Mexican, Italian, Beverages', 'rating': '4.5', 'price': '₹600 FOR TWO', 'link': 'the-bier-library-narayana-reddy-layout-koramangala-bangalore-101968'}
{'name': 'Desi Khana', 'type': 'North Indian, Punjabi, Chinese, Thalis', 'rating': '3.0', 'price': '₹150 FOR TWO', 'link': 'desi-khana-jakasandara-extension-hsr-layout-5th-sector-bangalore-101975'}
{'name': 'Terra Bites', 'type': 'Continental, American, Italian, Mexican', 'rating': '3.9', 'price': '₹700 FOR TWO', 'link': 'terra-bites-4th-block-koramangala-bangalore-102148'}
{'name': 'Tibbs Frank

In [3]:
# ratings_array
ratings_df = pd.DataFrame(ratings_array)
print(ratings_df)
ratings_df = ratings_df.drop(columns=['address'])
ratings_df

                            name  rating      address
0                      Go italia     4.1  Not in data
1                Fiero! Pizzeria     4.1  Not in data
2               The Bier Library     4.5  Not in data
3                     Desi Khana     3.0  Not in data
4                    Terra Bites     3.9  Not in data
..                           ...     ...          ...
864               Bansuri Sweets     3.9  Not in data
865             Upahara Darshini     4.3  Not in data
866        IceBreakers Ice Cream     3.4  Not in data
867                   Poha House     4.0  Not in data
868  Khan Saheb Grills and Rolls     3.9  Not in data

[869 rows x 3 columns]


name  rating
0                      Go italia     4.1
1                Fiero! Pizzeria     4.1
2               The Bier Library     4.5
3                     Desi Khana     3.0
4                    Terra Bites     3.9
..                           ...     ...
864               Bansuri Sweets     3.9
865             Upahara Darshini     4.3
866        IceBreakers Ice Cream     3.4
867                   Poha House     4.0
868  Khan Saheb Grills and Rolls     3.9

[869 rows x 2 columns]

In [4]:
ratings_df = ratings_df.sort_values(by = 'rating',ascending = False)
print(ratings_df[:10])

                       name  rating
788         SHAWARMA INDIAH     5.0
780    Biryani Bowl Company     4.9
713           Frosty Frills     4.9
204               Momo Bowl     4.8
786          BOMBAY 2 DELHI     4.8
664      Breakfast Junction     4.8
253              SOUR HOUSE     4.8
672               Bao to me     4.8
472  ITC Fabelle Chocolates     4.8
666          Iyengar's Cake     4.8


Problem 12 : Top 10 Restaurants with Maximum Ratings in Banglore

SHAWARMA INDIAH     5.0

Biryani Bowl Company     4.9

Frosty Frills     4.9

Momo Bowl     4.8

BOMBAY 2 DELHI     4.8

Breakfast Junction     4.8

SOUR HOUSE     4.8

Bao to me     4.8

ITC Fabelle Chocolates     4.8

Iyengar's Cake     4.8
